File: src/backend/Recipe.Api/Domain/Recipe.cs
namespace Domain;

public class Recipe
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }

    public string PrepTime { get; set; }

    public string CookTime { get; set; }

    public Ingredient MainIngredient { get; set; }

    public string Cuisine { get; set; }
    public ICollection<Ingredient> Ingredients { get; set; }
    public ICollection<string> Directions { get; set; }
}

public class Ingredient
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Quantity { get; set; }
}

File: src/frontend/src/features/recipe/types/recipe.ts
export interface Ingredient {
  id: string;
  name: string;
}

export interface Cuisine {
  id: string;
  name: string;
}

export interface Allergy {
  id: string;
  name: string;
}

export interface Recipe {
  id: string;
  name: string;
  description: string;
  ingredients: Ingredient[];
  cuisines: Cuisine[];
  allergies: Allergy[];
  preparationTime: number;
  cookingTime: number;
  servings: number;
  instructions: string[];
  imageUrl?: string;
}

export interface CreateRecipeRequest {
  name: string;
  description: string;
  ingredientIds: string[];
  cuisineIds: string[];
  allergyIds: string[];
  preparationTime: number;
  cookingTime: number;
  servings: number;
  instructions: string[];
  imageUrl?: string;
}
